source/fundamentals/logging.txt

```diff
@@ -7,37 +7,212 @@ Logging
+programmatically by specifying client options in the ``MongoClient``
+constructor.
 
-   const uri = "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority";
-   const client = new MongoClient(uri, { monitorCommands:true });
+.. note::
 
-   client.on('commandStarted', (event) => console.debug(event));
-   client.on('commandSucceeded', (event) => console.debug(event));
-   client.on('commandFailed', (event) => console.debug(event));
+   Because connection strings are often shared among a deployment of different
+   appications that might have different logging support, we do not recommend
+   using a connection string to configure logging.
+
+Environment Variables
+~~~~~~~~~~~~~~~~~~~~~
+
+You can configure logging for different components of the driver by specifying
+a severity level in one or more of the following environment variables:
+
+- ``MONGODB_LOG_ALL``: Logs all events
+- ``MONGODB_LOG_COMMAND``: Logs all commands sent to the server
+- ``MONGODB_LOG_TOPOLOGY``: Logs any changes to the cluster topology
+- ``MONGODB_LOG_SERVER_SELECTION``: Logs the server selection process
+- ``MONGODB_LOG_CONNECTION``: Logs all connection pool events
+- ``MONGODB_LOG_CLIENT``: Logs all client events
+
+If you don't specify any of the preceding environment variables, the driver uses
+the value of ``MONGODB_LOG_ALL``, which if not specified, is implicitly set to
+``off``.
+
+.. tip::
+
+   Logging at the command level is the most performance-heavy logging option
+   due to the frequency in which commands are sent to the server. Only
+   specify this option if necessary for your application.
+
+The following example sets the log level for all components to ``debug`` except
+for ``MONGODB_LOG_COMMAND``:
+
+.. code-block:: bash
+
+   export MONGODB_LOG_ALL="debug"
+   export MONGODB_LOG_COMMAND="off"
+
+Log Location
+````````````
+
+You can specify whether the driver logs to ``stderr`` or ``stdout`` by setting
+the ``MONGODB_LOG_PATH`` environment variable to ``"stderr"`` or ``"stdout"``,
+as shown in the following example:
+
+.. code-block:: bash
+
+   export MONGODB_LOG_PATH="stderr"
+
+By default, the driver logs to ``stdout``.
+
+Document Length
+```````````````
+
+The driver stringifies logged documents by using EJSON, which limits strings to
+1000 characters by default. You can specify a maximum document length for your
+logger by specifying the ``MONGODB_LOG_MAX_DOCUMENT`` environment variable. Set
+this variable to ``0`` to not perform truncation.
+
+The following example sets the maximum document length to 500 characters:
+
+.. code-block:: bash
+
+   export MONGODB_LOG_MAX_DOCUMENT_LENGTH=500
+
+Client Options
+~~~~~~~~~~~~~~
+
+You can configure logging programmatically by specifying client options in the
+``MongoClient`` constructor. Because client options take precedence over environment
+variables, only specify them in the client if you want to override any
+environment variables.
```

Comment:
```suggestion
``MongoClient`` constructor. Specifying client options takes precedence over environment
variables so take care to specify them in the client if you no longer want the driver to respond to environment variables.
```

Just a suggestion, but my intention behind the rewording here is to highlight that the user should acknowledge that they are opting into no longer getting a "no code change" logger, the main selling point. Because, once that is shipped it could be quite difficult to go back on that choice. 

Perhaps this is worthy of another "tip" call out, but I don't want to get overly prescriptive on how people should use this feature